core/src/sumeragi/main_loop.rs

```diff
@@ -592,7 +594,10 @@ pub fn run<F>(
         let span_for_sumeragi_cycle = span!(Level::TRACE, "Sumeragi Main Thread Cycle");
         let _enter_for_sumeragi_cycle = span_for_sumeragi_cycle.enter();
 
-        sumeragi.connect_peers(&state.current_topology);
+        if last_connect_peers_instant.elapsed().as_millis() > 1000 {
+            sumeragi.connect_peers(&state.current_topology);
+            last_connect_peers_instant = Instant::now();
+        }
```

Comment:
Can we instead of trying to connect peers in every loop iteration do it when `current_topology` is changed? (Seems to be after `refresh_at_new_block`)

Reply:
No because you need the peers to connect to each other even when nothing is happening.

Reply:
@SamHSmith, what do you mean by that?
As i see it with `connect_peers` we are sending bunch of `ConnectPeer`, `DisconnectPeer` messages and after processing on p2p side peers will be connected and we can send messages to them.
Why we need to restarting connection over and over again without any changes in topology?